tree/dataframe/src/RTreeColumnReader.cxx

```diff
@@ -0,0 +1,109 @@
+{
+   return fTreeValue->GetAddress();
+}
+
+ROOT::Internal::RDF::RTreeOpaqueColumnReader::RTreeOpaqueColumnReader(TTreeReader &r, std::string_view colName)
+   : fTreeValue(std::make_unique<ROOT::Internal::TTreeReaderOpaqueValue>(r, colName.data()))
+{
+}
+
+ROOT::Internal::RDF::RTreeOpaqueColumnReader::~RTreeOpaqueColumnReader() = default;
+
+void *ROOT::Internal::RDF::RTreeUntypedValueColumnReader::GetImpl(Long64_t)
+{
+   return fTreeValue->Get();
+}
+
+ROOT::Internal::RDF::RTreeUntypedValueColumnReader::RTreeUntypedValueColumnReader(TTreeReader &r,
+                                                                                  std::string_view colName,
+                                                                                  std::string_view typeName)
+   : fTreeValue(std::make_unique<ROOT::Internal::TTreeReaderUntypedValue>(r, colName, typeName))
+{
+}
+
+ROOT::Internal::RDF::RTreeUntypedValueColumnReader::~RTreeUntypedValueColumnReader() = default;
+
+void *ROOT::Internal::RDF::RTreeUntypedArrayColumnReader::GetImpl(Long64_t entry)
+{
+   if (entry == fLastEntry)
+      return &fRVec; // we already pointed our fRVec to the right address
+
+   auto &readerArray = *fTreeArray;
+   const auto readerArraySize = readerArray.GetSize();
+
+   // The reader could not read an array, signal this back to the node requesting the value
+   if (R__unlikely(readerArray.GetReadStatus() == ROOT::Internal::TTreeReaderValueBase::EReadStatus::kReadError))
+      return nullptr;
+
+   if (readerArray.IsContiguous() && !(fCollectionType == ECollectionType::kRVecBool)) {
+      if (readerArraySize > 0) {
+         // trigger loading of the contents of the TTreeReaderArray
+         // the address of the first element in the reader array is not necessarily equal to
+         // the address returned by the GetAddress method
+         RVec<Byte_t> rvec(readerArray.At(0), readerArraySize);
+         swap(fRVec, rvec);
+      } else {
+         RVec<Byte_t> emptyVec{};
+         swap(fRVec, emptyVec);
+      }
+   } else {
+      // The storage is not contiguous: we cannot but copy into the RVec
+#ifndef NDEBUG
+      if (!fCopyWarningPrinted && !(fCollectionType == ECollectionType::kRVecBool)) {
+         Warning("RTreeColumnReader::Get",
+                 "Branch %s hangs from a non-split branch. A copy is being performed in order "
+                 "to properly read the content.",
+                 readerArray.GetBranchName());
+         fCopyWarningPrinted = true;
+      }
+#else
+      (void)fCopyWarningPrinted;
+#endif
+      if (readerArraySize > 0) {
+         // Caching the value type size since GetValueSize might be expensive.
+         if (fValueSize == 0)
+            fValueSize = readerArray.GetValueSize();
+         assert(fValueSize > 0 && "Could not retrieve size of collection value type.");
+         // Array is not contiguous, make a full copy of it.
+         fRVec = RVec<Byte_t>{};
```

Comment:
```suggestion
         fRVec.clear();
```
I'm not sure if this makes a difference, but when you clear, the RVec might not need to reallocate.